executor/executor.go

```diff
@@ -1923,3 +1929,8 @@ func setRPCInterceptorOfExecCounterForTxn(vars *variable.SessionVars, snapshot k
 		snapshot.SetOption(kv.RPCInterceptor, vars.StmtCtx.KvExecCounter.RPCInterceptor())
 	}
 }
+
+func canWeakConsistencyRead(ctx sessionctx.Context) bool {
+	sessionVars := ctx.GetSessionVars()
+	return sessionVars.ConnectionID > 0 && sessionVars.ReadConsistency.IsWeak() && plannercore.IsAutoCommitTxn(ctx)
```

Comment:
Here I filter the internal queries by connection ID.

Reply:
maybe  isWeakConsistencyRead is more suitable